plugins/inputs/mongodb/mongodb.go

```diff
@@ -144,7 +144,14 @@ func (m *MongoDB) Start(telegraf.Accumulator) error {
 	return nil
 }
 
-func (m *MongoDB) Stop() {}
+// Stop disconnect mongo connections when stop or reload
+func (m *MongoDB) Stop() {
+	for _, server := range m.clients {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		_ = server.client.Disconnect(ctx)
```

Comment:
Should we maybe at least log an error if it occurs?
```suggestion
		if err := server.client.Disconnect(ctx); err != nil {
			m.Log.Errorf("disconnecting from %q failed: %w", server, err)
		}
```

Reply:
Thank you for correcting